src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -412,23 +412,142 @@ private static Buffer2DRegion<byte> TrimTransparentPixels(Buffer2D<byte> buffer,
         int bottom = int.MaxValue;
         int left = int.MaxValue;
         int right = int.MinValue;
-
-        // Run through th buffer in a single pass. Use variables to track the min/max values.
         int minY = -1;
         bool isTransparentRow = true;
+
+        // Run through the buffer in a single pass. Use variables to track the min/max values.
         for (int y = 0; y < buffer.Height; y++)
         {
             isTransparentRow = true;
             Span<byte> rowSpan = buffer.DangerousGetRowSpan(y);
+            ref byte rowPtr = ref MemoryMarshal.GetReference(rowSpan);
+            nint rowLength = (nint)(uint)rowSpan.Length;
+            nint x = 0;
+
+#if NET7_0_OR_GREATER
+            if (Vector128.IsHardwareAccelerated && rowLength >= Vector128<byte>.Count)
+            {
+                Vector256<byte> trimmableVec256 = Vector256.Create(trimmableIndex);
+
+                if (Vector256.IsHardwareAccelerated && rowLength >= Vector256<byte>.Count)
+                {
+                    do
+                    {
+                        Vector256<byte> vec = Vector256.LoadUnsafe(ref rowPtr, (nuint)x);
+                        Vector256<byte> notEquals = ~Vector256.Equals(vec, trimmableVec256);
+
+                        if (notEquals != Vector256<byte>.Zero)
```

Comment:
At the moment I don't have any idea on how to make this branchless.
`isTransparentRow` could be tracked in a vector, but `left` and `right` not, as there's a mismatch of vector-types, namely `byte` and `int`. 

A quite complicated approach would be to use `VectorXYZ<byte>` and track the left and right -- but just before these can overflow merge it back to the scalar `left`, `right` and start over. But I guess the book-keeping is more work, so I'm not sure if this is actually faster. For sure the code gets painful.